lib/active_merchant/billing/gateways/element.rb

```diff
@@ -0,0 +1,267 @@
+        request = build_soap_request do |xml|
+          xml.CreditCardSale(xmlns: "https://transaction.elementexpress.com") do
+            add_credentials(xml)
+            add_payment_method(xml, payment)
+            add_transaction(xml, money)
+            add_terminal(xml, options)
+            add_address(xml, options)
+          end
+        end
+
+        commit('CreditCardSale', request, money)
+      end
+
+      def authorize(money, payment, options={})
+        request = build_soap_request do |xml|
+          xml.CreditCardAuthorization(xmlns: "https://transaction.elementexpress.com") do
+            add_credentials(xml)
+            add_payment_method(xml, payment)
+            add_transaction(xml, money)
+            add_terminal(xml, options)
+            add_address(xml, options)
+          end
+        end
+
+        commit('CreditCardAuthorization', request, money)
+      end
+
+      def capture(money, authorization, options={})
+        trans_id, _ = split_authorization(authorization)
+        options.merge!({trans_id: trans_id})
+
+        request = build_soap_request do |xml|
+          xml.CreditCardAuthorizationCompletion(xmlns: "https://transaction.elementexpress.com") do
+            add_credentials(xml)
+            add_transaction(xml, money, options)
+            add_terminal(xml, options)
+          end
+        end
+
+        commit('CreditCardAuthorizationCompletion', request, money)
+      end
+
+      def refund(money, authorization, options={})
+        trans_id, _ = split_authorization(authorization)
+        options.merge!({trans_id: trans_id})
+
+        request = build_soap_request do |xml|
+          xml.CreditCardReturn(xmlns: "https://transaction.elementexpress.com") do
+            add_credentials(xml)
+            add_transaction(xml, money, options)
+            add_terminal(xml, options)
+          end
+        end
+
+        commit('CreditCardReturn', request, money)
+      end
+
+      def void(authorization, options={})
+        trans_id, trans_amount = split_authorization(authorization)
+        options.merge!({trans_id: trans_id, trans_amount: trans_amount, reversal_type: "Full"})
+
+        request = build_soap_request do |xml|
+          xml.CreditCardReversal(xmlns: "https://transaction.elementexpress.com") do
+            add_credentials(xml)
+            add_transaction(xml, trans_amount, options)
+            add_terminal(xml, options)
+          end
+        end
+
+        commit('CreditCardReversal', request, trans_amount)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<AccountID>).+?(</AccountID>))i, '\1[FILTERED]\2').
```

Comment:
:+1: for not scrubbing if it's not actually PCI-level sensitive - accountID can be really handy for debugging if you need to look something up in logs and send to support staff somewhere